### PR TITLE
Toyota: use platform config

### PR DESCRIPTION
**Pull request by sshane**
We should think more about grouping, it's nice that you can do that outside the enum creation for stuff like DBCs, common specs, and alike ADAS platforms. Now, you'll have to grep to find common platform details.

Do we want to completely get rid of sets/grouping? This seems like a slight regression for readability.

One change we can make is utilize `__post_init__` like we do for the `CarInfo`, however since everything is in `CAR`, `CAR` won't exist yet while initializing the configs. We can use a nonstandard init function and call it later but that doesn't seem that nice

### CODE REVIEW

File: selfdrive/car/toyota/values.py [hunk 1]
```diff
@@ -99,127 +59,278 @@ class ToyotaCarInfo(CarInfo):
...
+  def init_config(self):
+    if self.platform_str in TSS2_CAR:
+      self.dbc_dict = dbc_dict('toyota_nodsu_pt_generated', 'toyota_tss2_adas')
+    else:
+      if self.platform_str in (CAR.PRIUS, CAR.CHR, CAR.CAMRY, CAR.AVALON_2019):
+        self.dbc_dict = dbc_dict('toyota_nodsu_pt_generated', 'toyota_adas')
+      if self.platform_str in (CAR.AVALON, CAR.HIGHLANDER, CAR.RAV4H, CAR.SIENNA, CAR.LEXUS_IS, CAR.LEXUS_NX, CAR.LEXUS_RC, CAR.LEXUS_RX):
+        self.dbc_dict = dbc_dict('toyota_tnga_k_pt_generated', 'toyota_adas')
+      else:
+        self.dbc_dict = dbc_dict('toyota_new_mc_pt_generated', 'toyota_adas')
```
Comment (sshane) [inline selfdrive/car/toyota/values.py:73]: This is the kind of clean up I would hope we'd enable with this refactor (reduce copy pasting), @jnewb1 is this what you had in mind or no?

File: selfdrive/car/toyota/values.py [hunk 2]
```diff
@@ -99,127 +59,278 @@ class ToyotaCarInfo(CarInfo):
...
+    ],
+    specs=CarSpecs(mass=4305. * CV.LB_TO_KG, wheelbase=3.0, steerRatio=14.2, tireStiffnessFactor=0.444)
+  ),
+  AVALON = ToyotaPlatformConfig(
+    "TOYOTA AVALON 2016",
+    [
+      ToyotaCarInfo("Toyota Avalon 2016", "Toyota Safety Sense P"),
+      ToyotaCarInfo("Toyota Avalon 2017-18"),
+    ],
+    specs=CarSpecs(mass=0. * CV.LB_TO_KG, wheelbase=0, steerRatio=0, tireStiffnessFactor=0.444)
```
Comment (sshane) [inline selfdrive/car/toyota/values.py:92]: not done yet

File: selfdrive/car/tests/test_docs.py [hunk 3]
```diff
@@ -26,6 +27,12 @@ def test_generator(self):
     self.assertEqual(generated_cars_md, current_cars_md,
                      "Run selfdrive/car/docs.py to update the compatibility documentation")
 
+  def test_platforms(self):
```
Comment (sshane) [inline selfdrive/car/tests/test_docs.py:30]: should make new file

**Comment by jnewb1** [comment]
One thing I was thinking was using the "or" syntax, like

```
COROLLA = TSS2_CAR | PlatformConfig(
   ...
)
```

So you can skip anything set in TSS2_CAR

File: selfdrive/car/toyota/values.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/toyota/values.py:73]: I think these should be the flags that we set, like: https://github.com/commaai/openpilot/pull/31584

(TSS2, NODSU, NO_STOP_TIMER, etc etc)

Comment (sshane) [reply selfdrive/car/toyota/values.py:73]: I know we can use flags, but to me grouping seems much cleaner and more maintainable. The logic for group exclusivity is also out in the open

Comment (jnewb1) [reply selfdrive/car/toyota/values.py:73]: well the idea for the refactor was that all the car configuration is within one place, you modify one block of ~10 lines for your car and that's all you have to modify

Comment (jnewb1) [reply selfdrive/car/toyota/values.py:73]: > you could accidentally set multiple exclusive flags

well you can also add a car to two mutually exclusive groups right now, and we can probably setup something to enforce mutually exclusive groups

Comment (jnewb1) [reply selfdrive/car/toyota/values.py:73]: it's a bit of a trade off. Currently, if I want to search for how my corolla is controlled, I would have to search through the whole values.py to see what groups it's a part of, whereas if it's done with flags its more localized

it makes it harder to see which cars are in a group, but I think this is a less common use case

**Pull request merged**